src/sentry/incidents/logic.py

```diff
@@ -568,20 +565,17 @@ def create_alert_rule(
     """
     if monitor_type == AlertRuleMonitorTypeInt.ACTIVATED and not activation_condition:
         raise ValidationError("Activation condition required for activated alert rule")
-    if detection_type == AlertRuleDetectionType.DYNAMIC:
```

Comment:
This diff looks weird but I just combined the code for 2 checks for a dynamic rule